cms/djangoapps/contentstore/views/course.py

```diff
@@ -1566,20 +1663,42 @@ def group_configurations_detail_handler(request, course_key_string, group_config
             store.update_item(course, request.user.id)
             configuration = GroupConfiguration.update_usage_info(store, course, new_configuration)
             return JsonResponse(configuration, status=201)
+
         elif request.method == "DELETE":
             if not configuration:
                 return JsonResponse(status=404)
 
             # Verify that group configuration is not already in use.
-            usages = GroupConfiguration.get_usage_info(course, store)
-            if usages.get(int(group_configuration_id)):
-                return JsonResponse(
-                    {"error": _("This Group Configuration is already in use and cannot be removed.")},
-                    status=400
-                )
-
             index = course.user_partitions.index(configuration)
-            course.user_partitions.pop(index)
+            if configuration.scheme.name == 'random':
+                usages = GroupConfiguration.get_usage_info(course, store)
+                used = True if usages.get(int(group_configuration_id)) else False
+
+                if used:
+                    return JsonResponse(
+                        {"error": _("This Group Configuration is already in use and cannot be removed.")},
```

Comment:
Also, if the button label is "delete", then would it make more sense to refer consistently to "delete" (cannot be deleted) rather than introduce "remove"?